### ISSUE TITLE
Avalanche of unconfigured "fix" updates on Gradle updates

### ISSUE THREAD

**Issue by DanySK** [renovatebot/renovate#16845]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Since 3/4 days, renovate decided that most of my dependencies should switch from `chore` to `fix`, for no apparent reason. I did not change anything in the configuration, so I guess there has been a change in the default configuration I'm inheriting from. This also caused closed PRs to get reopened.
By looking at the docs, it now says that `fix` is the default for "npm production dependencies". Except, I received *many* `fix`-typed updates for JUnit and Kotest recently, that are most definitely not production dependencies, and they are not tracked by npm.
This caused in turn the release of many libraries / gradle plugins...

Was this desired behavior?
If so, what is the rationale of a `fix` commit (hence, a new patch release) on updates of the testing framework Gradle side?
I guess that all dependencies in `libs.versions.toml` has been interpreted as "production", but that is far from reality in most projects.
My feeling is that this is a behaviour designed for npm that slipped through other managers...

Here is the original config:
https://github.com/DanySK/renovate-config/blob/5b5c17f0d0d805968c77497ae495152815255c73/default.json#L1-L36

and here is an example of reopened pull request with an undesired change in commit type: 
* original: https://github.com/Protelis/Protelis/pull/712
* new pr: https://github.com/Protelis/Protelis/pull/764

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment]
@Churro I guess this is related to your recent addition of depType metadata to extracted Gradle dependencies combined with a preset which is in  config:base to use fix for production does and chore for dev dependencies. 

@DanySK I'm not sure this represents a bug although obviously we'd always prefer to avoid surprises or reopening of PRs

**Comment by Churro** [comment]
Correct. Renovate cannot know if dependencies specified in a gradle version catalog are production-used, as there is simply no annotation for that in a version catalog.

Until my commit, gradle deps had no depType set, making them always "chore" commits. This applied to production dependencies as well. I'd argue that most dependencies you specify with gradle or in a version catalog will end up in the compile or runtime class path.

Therefore, I don't think this is a bug

**Comment by DanySK** [comment]
Well, let's say that making JUnit update as "fix" is unusual at the very least.
Which are the available depTypes? I am willing to help with a better classification.

In Gradle, there are several dependency levels: api/implementation/runtimeOnly/compileOnly (the latter four also with testXyz version). That should be used to try to deduct the depType, or fall back to "chore".

**Comment by Churro** [comment]
The ones you mentioned are not dependency types but configuration types of the Java plugin. As such, they are instructions for gradle on how to use a dependency.

Personally, I also find it unituitive to use the _fix_ prefix for regular dependency update as per [conventional commits specification](https://www.conventionalcommits.org/en/v1.0.0/): _fix patches a bug in your codebase (this correlates with [PATCH](http://semver.org/#summary) in Semantic Versioning)._

To override this, I set `semanticCommitType: "ci"` in my `renovate.json`.

**Comment by rarkins** [comment]
FYI the reason for our convention of fix: is because of the view that any change to production dependencies should trigger a release (ie fix not chore)

**Comment by DanySK** [comment]
@Churro I disagree with you; if you imply that "most dependencies you specify with gradle or in a version catalog will end up in the compile or runtime classpath" (which is often not true) should mean that a new release gets triggered for every upgrade.
Most of them are not re-exported; they are implementation details, and should most definitely not generate a release.

I believe the current behaviour is *far* too aggressive, I maintain dozens of FOSS projects written in Kotlin/Java, and this change caused an avalanche of undesired patch releases.

Although the `api`/`implementation` is due to the Java plugin, it is *one* way to understand how a dependency is used. Because that's what we are discussing: the way a dependency is used.
If I use JUnit for testing, that's just a helper for me and in no way should trigger releases; if I'm writing a JUnit helper library, then it's a direct dependency and should get re-exported. It is the same dependency used differently.

My opinion is that everything should be a `chore` unless we can somewhat understand that it is actually not from the build structure. We are currently saying "everything but plugins is a core dependency", without making effort to understand whether it is actually not.

I entirely agree with you instead on the fact that `fix` should not be used for dependency updates. I believe that this is a clash between semantic commits and semantic release. I personally configured the latter with [this module](https://www.npmjs.com/package/semantic-release-preconfigured-conventional-commits), which uses scopes instead:
* `chore(deps)` are non-exported dependencies (e.g., the aforementioned JUnit case)
* `chore(core-deps)` are exported dependencies that should trigger a new PATCH
* `chore(api-deps)` are exported dependencies that taint the API, and thus should likely generate a MINOR release (for instance, I wrote a Gradle plugin that pre-configures Checkstyle/PMD/CPD/Spotbugs for Java, and any new release of any of these tools should trigger a minor).

Summing up, my opinions are:
1. It would be probably better if renovate did not use `fix` as default, regardless of the package manager
2. for Gradle specifically, it should at least try to understand how a library is being used, and probably if it can't understand it should default to `chore`
3. There is a bit of confusion out there on the mixture of conventional commits / semantic versioning / semantic release. A unifying effort would probably be useful.

**Comment by rarkins** [comment]
Renovate uses chore by default. You have opted into the config:base preset which alters that. Everything else is just N opinion. Choose yours and then set it as a central preset

**Comment by DanySK** [comment]
@rarkins yeah, it's no mystery that I expressed opinions :shrug: 
I'm already using a centralized preset and configuring it as I prefer; it still is interesting to discuss with other developers using Renovate.

**Comment by rarkins** [comment]
Sure, I just mean there's no right answer as to when to use semantic prefixes for dependency updates and you can configure it per your opinion. Closing this as it is working as expected
